### PR TITLE
Remove `airflow` and `spark` integrations

### PR DESCRIPTION
**Pull request by wslulciuc**
### Problem

The local build for Marquez fails since migrating to `Java17` (see #1819)

### Solution

This PR makes the following changes to fix the local build:

* Use the [`ubuntu-2004:202010-01`](https://circleci.com/docs/2.0/executor-intro/) image in CI for jvm all jobs and upgrade to using `adoptium` for `Java17` in [`.circleci/get-jdk17.sh`](https://github.com/MarquezProject/marquez/blob/main/.circleci/get-jdk17.sh). Note, the API docker images uses `adoptium` (#1770) as the base image, and this changes ensures all artifacts are using the same `Java17` distribution.
* Fixes the `pre-commit` cmd for apply spotless
* Adds `org.gradle.jvmargs` to the root `gradle.properties` file to fix backwards compatibility for `Java17`

This PR also removes the following modules no longer supported. I really didn't want to remove them in Marquez `0.21.0`, but they have been deprecated for a while, and backwards compatibility for `Java17` is greatly simplified. The CI build time is also reduced by `~6mins`:
* `marquez-airflow`
* `marquez-java`

### Checklist

- [x] You've [signed-off](https://github.com/MarquezProject/marquez/blob/main/CONTRIBUTING.md#sign-your-work) your work
- [ ] Your changes are accompanied by tests (_if relevant_)
- [ ] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [x] You've updated any relevant documentation (_if relevant_)
- [x] You've updated the [`CHANGELOG.md`](https://github.com/MarquezProject/marquez/blob/main/CHANGELOG.md#unreleased) with details about your change under the "Unreleased" section (_if relevant, depending on the change, this may not be necessary_)
- [ ] You've versioned your `.sql` database schema migration according to [Flyway's naming convention](https://flywaydb.org/documentation/concepts/migrations#naming) (_if relevant_)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -68,41 +68,25 @@ Marquez uses a _multi_-project structure and contains the following modules:
 * [`api`](https://github.com/MarquezProject/marquez/tree/main/api): core API used to collect metadata
 * [`web`](https://github.com/MarquezProject/marquez/tree/main/web): web UI used to view metadata
 * [`clients`](https://github.com/MarquezProject/marquez/tree/main/clients): clients that implement the HTTP [API](https://marquezproject.github.io/marquez/openapi.html)
-* [`integrations`](https://github.com/MarquezProject/marquez/tree/main/integrations): integrations with other systems (ex: [`Airflow`](https://github.com/MarquezProject/marquez/tree/main/examples/airflow))
```
Comment (collado-mike) [inline README.md:71]: Maybe worth calling out in the README that the integrations have been deleted and now exist only the OL repo

**Review by collado-mike — APPROVED** [review]
Let's do it! 😅

**Review by fm100 — APPROVED** [review]
👍

File: README.md (hunk 1 above)
Comment (wslulciuc) [reply README.md:71]: Good point!

Comment (wslulciuc) [reply README.md:71]: I did add the update to the changelog, but maybe we can have it in the README.md for a while 👍

**Pull request merged**